Team417_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team417_CENTERSTAGE/mechanisms/ArmTeleOp.java

```diff
@@ -0,0 +1,216 @@
+
+        armMotor.setPower(armPID(currentArmLocation, armMotor.getCurrentPosition(), armKp, armKi, armKd)); //Give the PID the location the arm should be at and location the arm is at.
+
+        lastArmProfileTime = currentTime; //The time the loop was competed
+    }
+
+    //Finds the smallest position in an array that is greater then current location of the mechanism. Returns 0 if no greater positions are found
+    //positionsArray: array of positions in encoder ticks, currentPosition: location of mechanism in encoder ticks from init location
+    //acceptableError: the amount of difference between the current location and a position in the array where it will still consider itself past the position
+    private double findNextPosition(double[] positionsArray, double currentPosition, double acceptableError) {
+        double nextPosition = 0;
+
+        for (int i = 0; i < positionsArray.length; i++) {
+            if (positionsArray[i] > currentPosition + acceptableError) {
+                nextPosition = positionsArray[i];
+                break;
+            }
+        }
+
+        return nextPosition;
+    }
+
+    //Finds the largest position in an array that is smaller then current location of the mechanism. Returns 0 if no smaller positions are found
+    //positionsArray: array of positions in encoder ticks, currentPosition: location of mechanism in encoder ticks from init location
+    //acceptableError: the amount of difference between the current location and a position in the array where it will still consider itself past the position
+    private double findLastPosition(double[] positionsArray, double currentPosition, double acceptableError) {
+        double lastPosition = 0;
+
+        for (int i = positionsArray.length - 1; i >= 0; i--) {
+            if (positionsArray[i] < currentPosition - acceptableError) {
+                lastPosition = positionsArray[i];
+                break;
+            }
+        }
+
+        return lastPosition;
+    }
+
+    public void armControl() {
+        double rStickSensitivity = 1, rStickSensitivityInsideRobot = rStickSensitivity / 3, rStickSensitivityOutsideRobot = rStickSensitivity; //How much moving the right stick will effect arm speed outside and inside the robot.
+        double[] ArmPositions = new double[] {BaseOpMode.ARM_MOTOR_MIN_POSITION, BaseOpMode.ARM_MOTOR_MAX_POSITION / 2.0,
+                BaseOpMode.ARM_MOTOR_MAX_POSITION * (3.0/4.0), BaseOpMode.ARM_MOTOR_MAX_POSITION}; //array of arm positions the dpad can move to.
+        double armCurrentLocation = armMotor.getCurrentPosition(); //The current distance the arm is from its init location
+        double rStickTilt = -gamepad2.right_stick_y; //The amount the right stick has been moved
+        final double rStickDeadZone = 0.1, armPosEpsilon = 25; //rStickDeadZone: the amount of space that is considered zero, armPosEpsilon: the amount of error between the current location and the goal location that is acceptable.
```

Comment:
These comments are awesome but they make the code lines really long so they're difficult to read with a normal sized window. Please consider putting the comments on the lines preceding the variables, and please respect the maximum line length of 100 characters that Android Studio recommends (that's the vertical grey line that it shows in the editor).

Reply:
Fixed that!